web/public/js/controllers/manage-dojo-users-controller.js

```diff
@@ -60,11 +62,243 @@ function cdManageDojoUsersCtrl($scope, $state, $q, cdDojoService, alertService,
+    icon: 'trophy',
+    title: $translate.instant('Award Badge'),
+    ngShow: function () {
+      return $scope.selectedItems.length === 1;
+    },
+    ngInclude: '/dojos/template/badges/award-form',
+    showAsAction: 'always'
+  };
+
+  function updateActions() {
+    $scope.actions = [{
+      icon: 'eye',
+      title: $translate.instant('View Profile'),
+      ngShow: function () {
+        return $scope.selectedItems.length === 1;
+      },
+      ngHref: function () {
+        var users = $scope.selectedItems;
+        if (users.length === 1) {
+          return '/profile/' + users[0].userData.id;
+        } else {
+          return null;
+        }
+      },
+      showAsAction: 'always'
+    }, {
+      icon: 'exchange',
+      title: $translate.instant('Change Role'),
+      ngShow: function () {
+        return $scope.selectedItems.length === 1;
+      },
+      ngClick: function () {
+        var user = $scope.selectedItems[0].userData;
+        var age = moment.utc().diff(moment(user.dob), 'years');
+        var roles = $scope.adultRoles;
+        if (age < 18) {
+          roles = $scope.youthRoles;
+        }
+        cdChooseRole({
+          roles: roles,
+          callback: function (role) {
+            return $q(function (resolve, reject) {
+              var hasPermission = false;
+              var userType = role.userType;
+              canUpdateUser(function (result) {
+                hasPermission = result;
+
+                if(hasPermission) {
+
+                  var userDojoLink = _.find(usersDojosLink, {userId:user.id});
+                  if(!userDojoLink.userTypes) userDojoLink.userTypes = [];
+                  var roles = [];
+                  if (age < 13) {
+                    roles.push('attendee-u13');
+                  } else if (age < 18) {
+                    roles.push('attendee-o13');
+                  }
+                  if (userType && userType.name) {
+                    roles.push(userType.name);
+                  }
+                  $scope.selectedItems[0].subCaption = userUtils.getTitleForUserTypes([userType.name]);
+                  userDojoLink.userTypes = roles;
+                  cdDojoService.saveUsersDojos(userDojoLink, function (response) {
+                    if(response.error) {
+                      reject($translate.instant(response.error));
+                    } else {
+                      resolve($translate.instant('User types successfully updated.'));
+                    }
+                  }, function (err) {
+                    reject($translate.instant('Error saving user type') + ' ' + err);
+                  });
+                } else {
+                  reject($translate.instant('You do not have permission to update user types'));
+                }
+              });
+            });
+          },
+          title: $translate.instant('Change Role')
+        });
+      },
+      showAsAction: 'ifRoom'
+    }, {
+      icon: 'minus-circle',
```

Comment:
Changing to a popup action would be more user friendly, as the confirmation would be closer to the click on the action bar (action bar at bottom, modal at top, not nice...)